src/components/ThirdPartyCollectionDetailPage/CollectionItem/CollectionItem.container.ts

```diff
@@ -0,0 +1,14 @@
+import { connect } from 'react-redux'
+import { push } from 'connected-react-router'
+import { openModal } from 'modules/modal/actions'
+import { MapStateProps, MapDispatch, MapDispatchProps } from './CollectionItem.types'
+import CollectionItem from './CollectionItem'
+
+const mapState = (): MapStateProps => ({})
+
+const mapDispatch = (dispatch: MapDispatch): MapDispatchProps => ({
+  onNavigate: path => dispatch(push(path)),
+  onOpenModal: (name, metadata) => dispatch(openModal(name, metadata))
```

Comment:
I think in this particular case it's not possible right? because I need to use two different modals
```ts
      onOpenModal('EditCollectionNameModal', { collection })
      onOpenModal('BuySlotModal', {})
```

Reply:
You can either give the component specific props, like `openBuySlotModal` or `openEditCollectionNameModal`, making the componente a bit more agnostic about the use of open modal or use it as is (with `onOpenModal`). I just pointed out a possible way of doing it to see what you though about it.
In the `CollectionItem.container` there's only one modal to be opened, right?